packages/firestore/src/api/database.ts

```diff
@@ -275,21 +230,9 @@ class FirestoreSettings {
       }
     }
 
-    validateNamedOptionalType(
-      'settings',
-      'boolean',
-      'experimentalForceLongPolling',
-      settings.experimentalForceLongPolling
-    );
     this.experimentalForceLongPolling =
       settings.experimentalForceLongPolling ?? DEFAULT_FORCE_LONG_POLLING;
 
-    validateNamedOptionalType(
-      'settings',
-      'boolean',
-      'experimentalAutoDetectLongPolling',
-      settings.experimentalAutoDetectLongPolling
-    );
     this.experimentalAutoDetectLongPolling =
       settings.experimentalAutoDetectLongPolling ??
```

Comment:
So given that we're not validating that `settings.experimentalAutoDetectLongPolling` is actually a boolean value, should we be accessing it with a forced coercion (i.e. as `!!settings.experimentalAutoDetectLongPolling`)?

Or is the idea that we've declared the type in typescript and all bets are off for users who aren't using our d.ts files?

Reply:
I originally decided against this approach, because it makes it impossible to turn a feature off it its default value is `true`. Since we have only ever been using opt-in features, this has not been a problem yet. I updated this code, which ensures that we have a boolean value once we exit this function (it also removes some code).